exchanges/kucoin/kucoin.go

```diff
@@ -0,0 +1,1864 @@
+	if chain != "" {
+		params["chain"] = chain
+	}
+	var resp []DepositAddress
+	return resp, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, kucoinGetDepositAddressV1, params, &resp)
+}
+
+// GetDepositList get deposit list items and sorted to show the latest first
+func (ku *Kucoin) GetDepositList(ctx context.Context, ccy, status string, startAt, endAt time.Time) ([]Deposit, error) {
+	params := make(map[string]interface{})
+	if ccy != "" {
+		params["currency"] = ccy
+	}
+	if status != "" {
+		params["status"] = status
+	}
+	if !startAt.IsZero() {
+		params["startAt"] = strconv.FormatInt(startAt.UnixMilli(), 10)
+	}
+	if !endAt.IsZero() {
+		params["endAt"] = strconv.FormatInt(endAt.UnixMilli(), 10)
+	}
+	resp := struct {
+		CurrentPage int64     `json:"currentPage"`
+		PageSize    int64     `json:"pageSize"`
+		TotalNum    int64     `json:"totalNum"`
+		TotalPage   int64     `json:"totalPage"`
+		Items       []Deposit `json:"items"`
+	}{}
+	return resp.Items, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, kucoinGetDepositList, params, &resp)
+}
+
+// GetHistoricalDepositList get historical deposit list items
+func (ku *Kucoin) GetHistoricalDepositList(ctx context.Context, ccy, status string, startAt, endAt time.Time) ([]HistoricalDepositWithdrawal, error) {
+	params := make(map[string]interface{})
+	if ccy != "" {
+		params["currency"] = ccy
+	}
+	if status != "" {
+		params["status"] = status
+	}
+	if !startAt.IsZero() {
+		params["startAt"] = strconv.FormatInt(startAt.UnixMilli(), 10)
+	}
+	if !endAt.IsZero() {
+		params["endAt"] = strconv.FormatInt(endAt.UnixMilli(), 10)
+	}
+	resp := struct {
+		CurrentPage int64                         `json:"currentPage"`
+		PageSize    int64                         `json:"pageSize"`
+		TotalNum    int64                         `json:"totalNum"`
+		TotalPage   int64                         `json:"totalPage"`
+		Items       []HistoricalDepositWithdrawal `json:"items"`
+	}{}
+	return resp.Items, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, kucoinGetHistoricalDepositList, params, &resp)
+}
+
+// GetWithdrawalList get withdrawal list items
+func (ku *Kucoin) GetWithdrawalList(ctx context.Context, ccy, status string, startAt, endAt time.Time) ([]Withdrawal, error) {
+	params := make(map[string]interface{})
+	if ccy != "" {
+		params["currency"] = ccy
+	}
+	if status != "" {
+		params["status"] = status
+	}
+	if !startAt.IsZero() {
+		params["startAt"] = strconv.FormatInt(startAt.UnixMilli(), 10)
+	}
+	if !endAt.IsZero() {
+		params["endAt"] = strconv.FormatInt(endAt.UnixMilli(), 10)
+	}
+	resp := struct {
+		CurrentPage int64        `json:"currentPage"`
+		PageSize    int64        `json:"pageSize"`
+		TotalNum    int64        `json:"totalNum"`
+		TotalPage   int64        `json:"totalPage"`
+		Items       []Withdrawal `json:"items"`
+	}{}
+	return resp.Items, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, kucoinWithdrawal, params, &resp)
```

Comment:
This is missing rate limiting

Reply:
Maybe I got it wrong but I see no descriptions on the rate limit for this endpoint and these are listed below.

Reply:
https://docs.kucoin.com/#get-withdrawals-list

> REQUEST RATE LIMIT
This API is restricted for each account, the request rate limit is 6 times/3s.